src/cli.rs

```diff
@@ -18,6 +18,9 @@ pub struct Cli {
   /// Project name. Defaults to target directory name
   #[structopt(short = "n", long = "name")]
   name: Option<String>,
+  /// Specify if a library template should be generated
+  #[structopt(short = "l", long = "lib")]
+  lib: Option<bool>,
```

Comment:
I believe `lib: bool` should work well enough. It defaults to `false`, and is only set to `true` if you pass `--lib` to the executable. This should make it slightly easier to access in the code.